src/scripts/uploadAutoCompleteLocations.js

```diff
@@ -0,0 +1,357 @@
+        return {
+          id: alternateNameId,
+          alternateOf: geonameId,
+          name: alternateName,
+          isPreferredName,
+        };
+      }
+
+      return null;
+    })
+    .filter(
+      (alternateName) => alternateName
+    );
+
+  console.info("Reading file: Hierarchy");
+  const hierachyData = readFileSync(
+    `${localDestinationPath.hierarchy}/hierarchy.txt`,
+    "utf8"
+  );
+  console.info("Parsing data: Location hierarchy");
+  const hierachyDataRows = hierachyData.split("\n");
+  const hierarchyIds = hierachyDataRows.map((hierachyRow) => {
+    const [locationParentId, locationChildId, _hierachyType] =
+      hierachyRow.split("\t");
+
+    return [locationParentId, locationChildId];
+  });
+
+  console.info("Reading file: All locations");
+  const locationData = readFileSync(
+    `${localDestinationPath.locations}/${DATA_COUNTRY_CODE}.txt`,
+    "utf8"
+  );
+
+  console.info("Parsing data: All locations");
+  const locationDataRows = locationData.split("\n");
+  const locationRowCount = locationDataRows.length;
+  const locationDataPopulated = locationDataRows.reduce(
+    (relevantLocations, location, rowIndex) => {
+      logProgress(rowIndex, locationRowCount);
+
+      const [
+        geonameId,
+        name,
+        _asciiname,
+        _alternatenames,
+        _latitude,
+        _longitude,
+        featureClass,
+        featureCode,
+        _countryCode,
+        _cc2,
+        admin1Code,
+        _admin2Code,
+        _admin3Code,
+        _admin4Code,
+        population,
+        _elevation,
+        _dem,
+        _timezone,
+        _modificationDate,
+      ] = location.split("\t") // lines are tab delimited
+
+      const isPopulatedPlaceOfInterest =
+        featureClass === allowedFeatureClass &&
+        allowedFeatureCodes.includes(featureCode);
+      const hasPopulation = Number(population) !== 0;
+
+      if (isPopulatedPlaceOfInterest && hasPopulation) {
+        const alternateNames = parsedAlternateNames.filter(
+          ({ alternateOf, name: alternateName }) =>
+            alternateOf === geonameId && alternateName !== name
+        );
+        const preferredName = alternateNames.find(
+          ({ isPreferredName }) => isPreferredName === "1"
+        );
+        const alternateNamesFinal = alternateNames.map((alternateName) => {
+          // Include the original name as an alternative name if we’ll use an
+          // alternate name that is the preferred name.
+          if (preferredName && preferredName.name === alternateName.name) {
+            return name;
+          }
+
+          return alternateName.name;
+        });
+
+        relevantLocations.push({
+          id: geonameId,
+          // switch names if an alternate name is the preferred location name
+          name: preferredName ? preferredName.name : name,
+          stateCode: admin1Code,
+          countryCode: "USA",
+          featureClass,
+          featureCode,
+          population,
```

Comment:
Not sure if it matters, but looks like `population` is still a string from the TSV file, not sure where/if it's being used downstream and if we want to cast it to Number:

```js
{ "id":"4905687",
  "name":"Peoria",
  "stateCode":"IL",
  "countryCode":"USA",
  "featureClass":"P",
  "featureCode":"PPLA2",
  "population":"115070" }
```